### PR TITLE
Add more retries for running ansible ping command (#2271)

### LINKED ISSUE #2271: [FEATURE REQUEST] Add more retries for running ansible ping command after terraform part
**Is your feature request related to a problem? Please describe.**

As for now we have [5 attempts](https://github.com/epiphany-platform/epiphany/blob/develop/core/src/epicli/cli/engine/ansible/AnsibleRunner.py#L62) to ping each machine after running terraform part, and that was good enough. But in the case of ARM testing, it sometimes happens that one of the machines is not ready yet, causing the build to fail with an error `Failed to connect to the host via ssh: System is booting up`.  This happens for ARM and from time to time when deploying an AWS CentOS cluster.

**Describe the solution you'd like**

We want to increase the number of retries, and/or add some timeout between attempts.

**Describe alternatives you've considered**
None.

**Additional context**
```
2021-04-29T12:17:24.6674123Z[38;21m12:17:24 INFO cli.engine.ansible.AnsibleCommand - Running: "p i n g"
2021-04-29T12:17:25.2760226Z[33;21m12:17:25 WARNING cli.engine.ansible.AnsibleCommand - [DEPRECATION WARNING]: ALLOW_WORLD_READABLE_TMPFILES option, moved to a per[0m
2021-04-29T12:17:25.2769340Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - plugin approach that is more flexible, use mostly the same config will work,[0m
2021-04-29T12:17:25.2771407Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - but now controlled from the plugin itself and not using the general constant.[0m
2021-04-29T12:17:25.2777892Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - instead. This feature will be removed from ansible-base in version 2.14.[0m
2021-04-29T12:17:25.2780000Z[33;21m12:17:25 WARNING cli.engine.ansible.AnsibleCommand - Deprecation warnings can be disabled by setting deprecation_warnings=False in[0m
2021-04-29T12:17:25.2786538Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - ansible.cfg.
2021-04-29T12:17:25.9007807Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - ec2-1-1-1-1.eu-west-1.compute.amazonaws.com | UNREACHABLE! => {
2021-04-29T12:17:25.9026436Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand -     "changed": false,
2021-04-29T12:17:25.9052247Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand -     "msg": "Failed to connect to the host via ssh: System is booting up. See pam_nologin(8)\nConnection closed by 1-1-1-1 port 22",
2021-04-29T12:17:25.9098088Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand -     "unreachable": true
2021-04-29T12:17:25.9125605Z[38;21m12:17:25 INFO cli.engine.ansible.AnsibleCommand - }
2021-04-29T12:17:26.7214338Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - ec2-2-2-2-2.eu-west-1.compute.amazonaws.com | SUCCESS => {
2021-04-29T12:17:26.7215248Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ansible_facts": {
2021-04-29T12:17:26.7216073Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -         "discovered_interpreter_python": "/usr/bin/python"
2021-04-29T12:17:26.7218141Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     },
2021-04-29T12:17:26.7219980Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "changed": false,
2021-04-29T12:17:26.7221003Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ping": "pong"
2021-04-29T12:17:26.7221652Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - }
2021-04-29T12:17:26.7330191Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - ec2-3-3-3-3.eu-west-1.compute.amazonaws.com | SUCCESS => {
2021-04-29T12:17:26.7332910Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ansible_facts": {
2021-04-29T12:17:26.7334246Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -         "discovered_interpreter_python": "/usr/bin/python"
2021-04-29T12:17:26.7336517Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     },
2021-04-29T12:17:26.7337230Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "changed": false,
2021-04-29T12:17:26.7337935Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ping": "pong"
2021-04-29T12:17:26.7341505Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - }
2021-04-29T12:17:26.7706346Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - ec2-4-4-4-4.eu-west-1.compute.amazonaws.com | SUCCESS => {
2021-04-29T12:17:26.7728796Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ansible_facts": {
2021-04-29T12:17:26.7738395Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -         "discovered_interpreter_python": "/usr/bin/python"
2021-04-29T12:17:26.7746751Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     },
2021-04-29T12:17:26.7755314Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "changed": false,
2021-04-29T12:17:26.7763606Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ping": "pong"
2021-04-29T12:17:26.7773837Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - }
2021-04-29T12:17:26.7787028Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - ec2-5-5-5-5.eu-west-1.compute.amazonaws.com | SUCCESS => {
2021-04-29T12:17:26.7798354Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ansible_facts": {
2021-04-29T12:17:26.7813285Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -         "discovered_interpreter_python": "/usr/bin/python"
2021-04-29T12:17:26.7814724Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     },
2021-04-29T12:17:26.7815857Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "changed": false,
2021-04-29T12:17:26.7824044Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand -     "ping": "pong"
2021-04-29T12:17:26.7830109Z[38;21m12:17:26 INFO cli.engine.ansible.AnsibleCommand - }
2021-04-29T12:17:26.8597782Z[31;21m12:17:26 ERROR cli.engine.ansible.AnsibleCommand - Error running: "ansible -m ping -i /shared/build/armawscentcanal/inventory all"
2021-04-29T12:17:26.8602510Z[33;21m12:17:26 WARNING cli.engine.ansible.AnsibleCommand - Retry running task: 5/5
2021-04-29T12:17:36.8708875Z[38;21m12:17:36 INFO cli.engine.ansible.AnsibleRunner - Run done in 74685ms
2021-04-29T12:17:36.8709791Z[31;21m12:17:36 ERROR epicli - Failed running task after 5 retries
```

---

**DoD checklist**

* [x] Changelog updated
* [x] COMPONENTS.md updated / doesn't need to be updated
* [ ] Feature has automated tests
* [ ] Automated tests passed (QA pipelines)
  * [x] apply
  * [x] upgrade
* [x] Idempotency tested
* [x] Documentation added / updated / doesn't need to be updated
* [x] All conversations in PR resolved
* [x] Solution meets requirements and is done according to design doc
* [x] Usage compliant with license
* [x] Backport tasks created / doesn't need to be backported

### PR DESCRIPTION
**Pull request by sbbroot**
* Add optional epicli argument `ping-retries` for apply and upgrade

### CODE REVIEW

File: core/src/epicli/cli/engine/ApplyEngine.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ def __init__(self, input_data):
         self.infrastructure_docs = []
         self.manifest_docs = []
 
+        self.__ping_retries: int = input_data.ping_retries
```
Comment (seriva) [inline core/src/epicli/cli/engine/ApplyEngine.py:38]: Apply __ everywhere if ur bored;)

**Review by seriva — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Review by erzetpe — CHANGES_REQUESTED** [review]
Please change the the branch.

**Comment by seriva** [comment]
> Please change the the branch.

Yeah Pawo is right, this is not for 1.2 anymore. So put it on a separate branch and move the changelog to a new 1.3 file pls.

File: core/src/epicli/cli/epicli.py [hunk 2]
```diff
@@ -202,6 +202,8 @@ def apply_parser(subparsers):
                             help='Password that will be used to encrypt build artifacts.')
     optional.add_argument('--profile-ansible-tasks', dest='profile_ansible_tasks', action="store_true",
                             help='Enable Ansible profile_tasks plugin for timing tasks. (developer/debug option)')
+    optional.add_argument('--ping-retries', dest='ping_retries', type=int, required=False, action='store', default=5,
```
Comment (to-bar) [inline core/src/epicli/cli/epicli.py:205]: IMO it would be nice to have arguments in alphabetic order but this can be done next time.

**Review by to-bar — DISMISSED** [review]

**Comment by to-bar** [comment]
> 
> 
> > Please change the the branch.
> 
> Yeah Pawo is right, this is not for 1.2 anymore. So put it on a separate branch and move the changelog to a new 1.3 file pls.

IMO the target branch is ok (`develop`). We will need to create `v1.2` for releasing 1.2.0.  I would opt for keeping this PR open until the new branch is created. However the entry in changelog should be moved to the new file (`CHANGELOG-1.3.md`).

**Comment by mkyc** [comment]
Confirm with @przemyslavic before merging that he is ok with it.

**Comment by seriva** [comment]
> Please change the the branch.

As we discussed last week, its a minor change, so if @przemyslavic  is good with it I am good for merging for 1.2.

**Review by seriva — DISMISSED** [review]

**Review by przemyslavic — APPROVED** [review]

**Review by erzetpe — APPROVED** [review]

**Pull request merged**